pkg/util/util.go

```diff
@@ -341,3 +346,18 @@ func ClearMacAddress(vmiSpec *kvv1.VirtualMachineInstanceSpec) {
 		vmiSpec.Domain.Devices.Interfaces[i].MacAddress = ""
 	}
 }
+
+// no special meaning, randomly generated as done in Kubevirt.
+const magicUUID = "a8a1b7a9-d8c3-4b2f-9733-7a3fbb1f1234"
+
+var firmwareUUIDns = uuid.MustParse(magicUUID)
+
+func ShouldGenerateNewFirmwareUUID(restore *velerov1.Restore) bool {
+	return metav1.HasLabel(restore.ObjectMeta, GenerateNewFirmwareUUIDLabel)
+}
+
+// GenerateNewFirmwareUUID generates a new firmware UUID for the restored VM, ensuring it will be different than the one
+// Kubevirt generates for the original VM by using a different magicUUID and hashing name+namespace.
+func GenerateNewFirmwareUUID(vmiSpec *kvv1.VirtualMachineInstanceSpec, name, namespace string) {
+	vmiSpec.Domain.Firmware.UUID = types.UID(uuid.NewSHA1(firmwareUUIDns, []byte(name+namespace)).String())
```

Comment:
Dis you get this code from kubevirt? Shouldn't there be `/` between namespace and name? Otherwise `hash(foo, bar) = hash(f, oobar)`?

Reply:
See what they do here: https://github.com/kubevirt/kubevirt/pull/13150

Reply:
ACK

Reply:
Should've looked at this sooner: https://github.com/kubevirt/community/pull/347/

namespace/name was abandoned for object uid. Unfortunately we cannot use that here so may as well just generate new random uids

Reply:
Interesting, but we can't do much until it merges. Are we still ok with this implementation?

Reply:
I'm not against generating a random one but currently I'm hashing `name/namespace/UID`, isn't that enough to secure unique IDs per VM?

Reply:
No because if a VM with same name/namespace is deleted+recreated it will have the same id

Reply:
gotcha